lib/model/model.go

```diff
@@ -1071,17 +1071,17 @@ func (m *model) handleIndex(deviceID protocol.DeviceID, folder string, fs []prot
 	m.fmut.RUnlock()
 
 	if !existing {
-		l.Warnf("%v for nonexistent folder %q", op, folder)
-		panic("handling index for nonexistent folder")
+		l.Infof("%v for nonexistent folder %q", op, folder)
+		return errFolderMissing
 	}
 
 	if running {
 		defer runner.SchedulePull()
 	} else if update {
 		// Runner may legitimately not be set if this is the "cleanup" Index
 		// message at startup.
```

Comment:
Unrelated: I believe this comment (and the entire if running block) is incorrect: I don't know what "cleanup" Index it is referring to and no matter what it is, a runner must always be set up if `existing == true`.

Reply:
I think this is legacy. Back in the day we did `model.Index(someFolder, someDevice, nil)` to drop index data for that device/folder in main.

Reply:
I don't see any such call so I removed that check and simplified

Reply:
... but of course we do so all the time from tests...